shell/platform/common/client_wrapper/include/flutter/plugin_registrar.h

```diff
@@ -54,7 +54,7 @@ class PluginRegistrar {
   void AddPlugin(std::unique_ptr<Plugin> plugin);
 
  protected:
-  FlutterDesktopPluginRegistrarRef registrar() { return registrar_; }
+  FlutterDesktopPluginRegistrarRef registrar() const { return registrar_; }
```

Comment:
Should we be conservative about applying `const` on our public APIs?

Reply:
I've heard two lines of thought on this:
* Is the method mutating the external state of the object? No: apply `const`.
* Is the above true, but you're returning a non-const ref to internal state such that the user of the return value can mutate the state of your object? Yes: don't apply it.

I tend to lean towards the first option since if you really want to prevent that then you can quality the return type with const. If the C++ language designers had wanted the second interpretation, they had all the info they needed to prevent you from returning a non-const bit of state from a const method and could have required you to, for example, mark the field `mutable` if so, but they didn't.

i.e. I say go for it (hence why I didn't comment on the last patch that did this)

Reply:
> imagine that for whatever reason we need to lazily build the registrar the first time it is accessed

This is exactly what the `mutable` keyword is for, so I think we'd be fine in that case.

Reply:
Ah didn't know about `mutable`, that addresses my concern! Thanks for the pointer, will close